tests/cdi/cdi_test.go

```diff
@@ -0,0 +1,135 @@
+package cdi_test
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// template parameters
+const (
+	pvcEPHTTPNOAUTHURL  = "https://download.cirros-cloud.net/0.4.0/cirros-0.4.0-x86_64-disk.img"
+	pvcName        = "golden-pvc"
+	vmName         = "test-vm"
+	rawPVCFilePath = "tests/cdi/golden-pvc.yml"
+	rawVMFilePath  = "tests/cdi/test-vm.yml"
+	sepP = "-p"
+)
+
+var KubeVirtOcPath = ""
+
+var _ = Describe("CDI populate pvc create binding, running vm", func() {
+	flag.Parse()
+	SkipIfNoOc()
+	dstPVCFilePath := "/tmp/test-pvc"
+	dstVMFilePath := "/tmp/test-vm"
+
+	overrideTemplateFromParameters := func(srcFilePath, dstFilePath string, params ...string) string {
+		By(fmt.Sprintf("Overriding template from %s to %s", srcFilePath, dstFilePath))
+		args := []string{"process", "-f", srcFilePath}
+		args = append(args, params...)
+		out, err := RunOcCommand(args...)
+		Expect(err).ToNot(HaveOccurred())
+		filePath, err := WriteJson(dstFilePath, out)
+		Expect(err).ToNot(HaveOccurred())
+		return filePath
+	}
+
+	createResourceWithFilePath := func(resourceName, resourceType, filePath string) {
+		By(fmt.Sprintf("Creating %s:%s from JSON file via oc-create command", resourceType, resourceName))
+		_, err := RunOcCommand("create", "-f", filePath)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() bool {
+			out, err := RunOcCommand("get", resourceType)
+			Expect(err).ToNot(HaveOccurred())
+			return strings.Contains(out, resourceName)
+		}, time.Duration(2)*time.Minute).Should(BeTrue(), "Timed out waiting for ovm to appear")
+	}
+
+	waitUntilResourceReady := func(resourceName, resourceType, query, expectOut string) {
+		By(fmt.Sprintf("Wait until %s:%s ready", resourceType, resourceName))
+		Eventually(func() bool {
+			//oc get resource with -o jsonpath query
+			out, err := RunOcCommand("get", resourceType, resourceName, query)
+			Expect(err).ToNot(HaveOccurred())
+			return strings.Contains(out, expectOut)
+		}, time.Duration(2)*time.Minute).Should(BeTrue(), "Timed out waiting for vm to appear")
+	}
+
+	Context("will success", func() {
+		It("create pvc should success", func() {
```

Comment:
"should create the PVC successfully"

Reply:
done, thanks for correct my grammatical errors, I love this.